test/Mono.Linker.Tests.Cases/RequiresCapability/RequiresCapability.cs

```diff
@@ -809,18 +897,279 @@ class AccessThroughLdToken
+				public virtual string VirtualPropertyAnnotationInProperty { get; set; }
+			}
+
+			class BaseClassWithoutRequires
+			{
+				public virtual void VirtualMethod ()
+				{
+				}
+
+				public virtual string VirtualPropertyAnnotationInAccesor { get; set; }
+
+				public virtual string VirtualPropertyAnnotationInProperty { get; set; }
+			}
+
+			class DerivedClassWithRequires : BaseClassWithoutRequires
+			{
+				[RequiresUnreferencedCode ("Message")]
+				[RequiresAssemblyFiles ("Message")]
+				[ExpectedWarning ("IL2046", "DerivedClassWithRequires.VirtualMethod()", "BaseClassWithoutRequires.VirtualMethod()")]
+				[ExpectedWarning ("IL3003", "DerivedClassWithRequires.VirtualMethod()", "BaseClassWithoutRequires.VirtualMethod()", ProducedBy = ProducedBy.Analyzer)]
+				public override void VirtualMethod ()
+				{
+				}
+
+				private string name;
+				public override string VirtualPropertyAnnotationInAccesor {
+					[ExpectedWarning ("IL2046", "DerivedClassWithRequires.VirtualPropertyAnnotationInAccesor.get", "BaseClassWithoutRequires.VirtualPropertyAnnotationInAccesor.get")]
+					[ExpectedWarning ("IL3003", "DerivedClassWithRequires.VirtualPropertyAnnotationInAccesor.get", "BaseClassWithoutRequires.VirtualPropertyAnnotationInAccesor.get", ProducedBy = ProducedBy.Analyzer)]
+					[RequiresUnreferencedCode ("Message")]
+					[RequiresAssemblyFiles ("Message")]
+					get { return name; }
+					set { name = value; }
+				}
+
+				[RequiresAssemblyFiles ("Message")]
+				[ExpectedWarning ("IL3003", "DerivedClassWithRequires.VirtualPropertyAnnotationInProperty", "BaseClassWithoutRequires.VirtualPropertyAnnotationInProperty", ProducedBy = ProducedBy.Analyzer)]
+				public override string VirtualPropertyAnnotationInProperty { get; set; }
+			}
+
+			class DerivedClassWithoutRequires : BaseClassWithRequires
+			{
+				[ExpectedWarning ("IL2046", "DerivedClassWithoutRequires.VirtualMethod()", "BaseClassWithRequires.VirtualMethod()")]
+				[ExpectedWarning ("IL3003", "DerivedClassWithoutRequires.VirtualMethod()", "BaseClassWithRequires.VirtualMethod()", ProducedBy = ProducedBy.Analyzer)]
+				public override void VirtualMethod ()
+				{
+				}
+
+				private string name;
+				public override string VirtualPropertyAnnotationInAccesor {
+					[ExpectedWarning ("IL2046", "DerivedClassWithoutRequires.VirtualPropertyAnnotationInAccesor.get", "BaseClassWithRequires.VirtualPropertyAnnotationInAccesor.get")]
+					[ExpectedWarning ("IL3003", "DerivedClassWithoutRequires.VirtualPropertyAnnotationInAccesor.get", "BaseClassWithRequires.VirtualPropertyAnnotationInAccesor.get", ProducedBy = ProducedBy.Analyzer)]
+					get { return name; }
+					set { name = value; }
+				}
+
+				[ExpectedWarning ("IL3003", "DerivedClassWithoutRequires.VirtualPropertyAnnotationInProperty", "BaseClassWithRequires.VirtualPropertyAnnotationInProperty", ProducedBy = ProducedBy.Analyzer)]
+				public override string VirtualPropertyAnnotationInProperty { get; set; }
```

Comment:
Why is this for assembly files only and not for unreferenced code as well?

Reply:
You cannot place RUC on a property, but you can do it in RAF

Reply:
Hmm that's a curious limitation... can you please create an issue somewhere to make this consistent? Unless there's a reason (and I just forgot).

Reply:
What about events?

Reply:
I think it's the same deal... In theory there's no reason for these two attributes to have different behavior (at least I'm not aware of one). It's fine if they're not identical right now, but I think they should be eventually.